application/controllers/ListController.php

```diff
@@ -0,0 +1,342 @@
+        $this->view->stats = $stats;
+        if (strpos($this->params->get('host_name', '*'), '*') === false) {
+            $this->view->showHost = false;
+        } else {
+            $this->view->showHost = true;
+        }
+    }
+    /**
+     * List service groups
+     */
+    public function servicegroupsAction()
+    {
+        $this->addTitleTab(
+            'servicegroups',
+            $this->translate('Service Groups'),
+            $this->translate('List service groups')
+        );
+
+        $this->setAutorefreshInterval(12);
+
+        $serviceGroups = $this->backend->select()->from('servicegroupsummary', array(
+            'servicegroup_alias',
+            'servicegroup_name',
+            'services_critical_handled',
+            'services_critical_unhandled',
+            'services_ok',
+            'services_pending',
+            'services_total',
+            'services_unknown_handled',
+            'services_unknown_unhandled',
+            'services_warning_handled',
+            'services_warning_unhandled'
+        ));
+
+        $this->setupPaginationControl($serviceGroups);
+        $this->setupSortControl(array(
+            'servicegroup_alias'    => $this->translate('Service Group Name'),
+            'services_severity'     => $this->translate('Severity'),
+            'services_total'        => $this->translate('Total Services')
+        ), $serviceGroups);
+        $this->filterQuery($serviceGroups);
+        $this->setupLimitControl();
+
+        $this->view->serviceGroups = $serviceGroups;
+    }
+
+    /**
+     * List service groups
+     */
+    public function servicegroupGridAction()
+    {
+        $this->addTitleTab(
+            'servicegroup-grid',
+            $this->translate('Service Group Grid'),
+            $this->translate('Show the Service Group Grid')
+        );
+
+        $this->setAutorefreshInterval(15);
+
+        $serviceGroups = $this->backend->select()->from('servicegroupsummary', array(
+            'servicegroup_alias',
+            'servicegroup_name',
+            'services_critical_handled',
+            'services_critical_unhandled',
+            'services_ok',
+            'services_pending',
+            'services_total',
+            'services_unknown_handled',
+            'services_unknown_unhandled',
+            'services_warning_handled',
+            'services_warning_unhandled'
+        ));
+        $this->applyRestriction('monitoring/filter/objects', $serviceGroups);
+        $this->filterQuery($serviceGroups);
+
+        $this->setupSortControl(array(
+            'servicegroup_alias'    => $this->translate('Service Group Name'),
+            'services_severity'     => $this->translate('Severity'),
+            'services_total'        => $this->translate('Total Services')
+        ), $serviceGroups, ['services_severity' => 'desc']);
+
+        $this->view->serviceGroups = $serviceGroups;
+    }
+
+    public function servicegridAction()
```

Comment:
I think this should be a part of the `ServicesController`. (i.e. `icingadb/services/grid`)

This has the advantage that you can use nearly (minus e.g. the view-mode-switcher) the same instructions as the `indexAction` to set up the controls. (Because the grid is also using `Icinga\Module\Icingadb\Model\Service` as base model)